doc/reference/trd-legal.md

```diff
@@ -0,0 +1,188 @@
+microcontroller-class devices. The Tock project further believes it is
+important to enable and support widespread adoption of safe, secure, and
+efficient software. Tock also seeks to be an open and inclusive project, and
+Tock welcomes contributions from any individuals or entities wishing to
+improve the safety, security, reliability, efficiency, or usability of Tock
+and the Tock ecosystem.
+
+The intent of these policies is to best satisfy the needs of all stakeholders
+in the Tock ecosystem.
+
+
+2. Licensing
+============
+
+Tock’s primary licensing goal is to ensure ease of use for Tock project elements.
+For this reason, Tock offers its resources under the minimally encumbered MIT
+license and under the Apache2 license for those entities concerned about possible
+litigation issues.
+
+All software artifacts under the umbrella of the Tock project are
+dual-licensed as Apache License, Version 2.0 (LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0) or MIT license (LICENSE-MIT or
+http://opensource.org/licenses/MIT).
+
+All contributions to Tock projects MUST be licensed under these terms.
+
+
+3. Copyright
+============
+
+Entities contributing resources to open-source projects often require attribution
+to recognize their efforts. Copyright is a common means to provide this. For
+downstream users, the license terms of Tock ensures unencumbered use.
+
+Copyrights in Tock projects are retained by their contributors. No
+copyright assignment is required to contribute to Tock projects.
+
+Artifacts in the Tock project MAY include explicit copyright notices.
+Substantial updates to an artifact MAY add additional copyright notices
+to an artifact. Copyright notices MUST NOT be removed by any party except
+the original copyright holder, and removal is done solely at their discretion.
+
+For full authorship information, see the version control history.
+
+
+4. Implementation
+=================
+
+All code artifacts in Tock projects MUST include a license notice.
+
+Code artifacts MAY include copyright notice(s). Copyright notices MUST
+include a year. Newer copyright notices MUST be placed after existing
+copyright notices. If non-trivial updates are performed by an original
+copyright author, they MAY amend the year(s) indicated on their existing
+copyright statement or MAY add an additional copyright line, at their
+discretion.
+
+
+4.1 Format
+----------
+
+License and copyright information MUST appear near the top of a file.
+Generally, license and copyright information SHOULD be the first lines
+in a file, however exceptions may be made where technically necessary
+(e.g. to accommodate the shebang in a shell script).
+
+License and copyright information MUST have exactly one (1) blank line
+separating it from any other content in the file.
+
+License information MUST come before copyright information.
+There MUST NOT be any whitespace between lines in the license/copyright block.
```

Comment:
I think we want to avoid specifying this at such a low level. E.g., the standard format of a BSD license in a comment block has lines of whitesapce between the enumerated elements. That is, remove this MUST NOT.